### PR TITLE
Update Hamburger design

### PR DESCRIPTION
**Pull request by connor-baer**
## Purpose

The new navigation components require an update to the Hamburger design: 

_Before_

Screenshot 2021-08-03 at 17 29 52

_After_

Screenshot 2021-08-03 at 17 29 43

While working on the changes, I also noticed that the Button's `kilo` size is slightly too tall.

## Approach and changes

- Tweaked the Hamburger design and aligned its height with the Button component.
- Reduced the height of the Button's `kilo` size from 34px to 32px to align it with the other components.

## Definition of done

* [x] Development completed
* [x] Reviewers assigned
* [x] Unit and integration tests
* [x] Meets minimum browser support
* [x] Meets accessibility requirements

### CODE REVIEW

File: packages/circuit-ui/components/Button/__snapshots__/Button.spec.tsx.snap [hunk 1]
```diff
@@ -280,7 +280,7 @@ exports[`Button styles should render a kilo button 1`] = `
   background-color: #FFF;
   border-color: #999;
   color: #000;
-  padding: 4px calc(16px - 1px);
+  padding: calc(4px - 1px) calc(16px - 1px);
```
Comment (robinmetral) [inline packages/circuit-ui/components/Button/__snapshots__/Button.spec.tsx.snap:283]: 😢

**Review by robinmetral — APPROVED** [review]
Nice! 🙌

File: packages/circuit-ui/components/Button/__snapshots__/Button.spec.tsx.snap (hunk 1 above)
Comment (amelako) [reply packages/circuit-ui/components/Button/__snapshots__/Button.spec.tsx.snap:283]: 😩

**Review by amelako — APPROVED** [review]
🥳

**Pull request merged**